compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/types/Types.java

```diff
@@ -0,0 +1,201 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.compiler.api.types;
+
+import io.ballerina.compiler.api.impl.SymbolFactory;
+import io.ballerina.compiler.api.impl.symbols.TypesFactory;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import org.ballerinalang.model.elements.PackageID;
+import org.ballerinalang.model.symbols.SymbolKind;
+import org.ballerinalang.model.symbols.SymbolOrigin;
+import org.wso2.ballerinalang.compiler.PackageCache;
+import org.wso2.ballerinalang.compiler.semantics.model.Scope;
+import org.wso2.ballerinalang.compiler.semantics.model.SymbolEnv;
+import org.wso2.ballerinalang.compiler.semantics.model.SymbolTable;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BPackageSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.Symbols;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.Names;
+import org.wso2.ballerinalang.util.Flags;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Represents the set of built-in and user-defined types available in a single semantic context.
+ *
+ * @since 2.0.0
+ */
+
+public class Types {
+
+    private static final CompilerContext.Key<Types> TYPES_KEY = new CompilerContext.Key<>();
+    private final SymbolFactory symbolFactory;
+    private final SymbolTable symbolTable;
+    private final PackageCache packageCache;
+
+    public final TypeSymbol BOOLEAN;
+    public final TypeSymbol INT;
+    public final TypeSymbol FLOAT;
+    public final TypeSymbol DECIMAL;
+    public final TypeSymbol STRING;
+    public final TypeSymbol NIL;
+    public final TypeSymbol XML;
+    public final TypeSymbol ERROR;
+    public final TypeSymbol FUNCTION;
+    public final TypeSymbol FUTURE;
+    public final TypeSymbol TYPEDESC;
+    public final TypeSymbol HANDLE;
+    public final TypeSymbol STREAM;
+    public final TypeSymbol ANY;
+    public final TypeSymbol ANYDATA;
+    public final TypeSymbol NEVER;
+    public final TypeSymbol READONLY;
+    public final TypeSymbol JSON;
+    public final TypeSymbol BYTE;
+    public final TypeSymbol COMPILATION_ERROR;
+    public final CompilerContext context;
```

Comment:
What's the need for making the compiler context final? This should not be exposed at all.